utils/binary/read.go

```diff
@@ -25,6 +25,15 @@ func Read(r io.Reader, data ...interface{}) error {
 
 // ReadUntil reads from r untin delim is found
 func ReadUntil(r io.Reader, delim byte) ([]byte, error) {
+	if bufr, ok := r.(*bufio.Reader); ok {
```

Comment:
Can you move this to a function called `ReadUntilFromBufioReader` or something like that, and call it from here?

Reply:
Yeah, will push shortly with a test for it.

The only problem with this approach is that bufio.Reader satisfies io.Reader too, so if you pass a bufio.Reader to it, it's not going to be immediately obvious you're getting slower behaviour.

There's very few places using ReadUntil, so it's not an issue now. But it's something to look out for in the future and to ensure that if the reader is buffered, they're using the correct variant.

Reply:
I mean call one from the other, while the new behavior is public

Reply:
```
if bufr, ok := r.(*bufio.Reader); ok {
	return ReadUntilFromBufioReader(bufr)
}
```

Reply:
Done! Sorry about that, re-reading your comment, it was clear enough. My bad.